### ISSUE TITLE
xmldom parserError  - How to debug

### ISSUE THREAD

**Issue by ramiboutas** [Instawork/hyperview#787]
Is there any way to check what the HXML client receives? I am getting the xmldom ParserError all the times. I assume the the client receives an empty file or something.

I set up the Content-Type header to `application/vnd.hyperview+xml` in my Django http reponse.

**Comment by ramiboutas** [comment]
I followed this tutorial: https://hypermedia.systems/building-a-contacts-app-with-hyperview/

**Comment by adamstep** [comment]
Hi @ramiboutas , one way to see the HXML received by the client is to use a proxy, on the Mac I use Proxyman. If you share your code I may be able to help understand what's going on.

Also, the code from the book is available online as well, you can check out these branches:

- Basic contacts app implementation: https://github.com/adamstep/contact-app/commits/adam/hv3/
- Add support for phone calls and email: https://github.com/adamstep/contact-app/commits/adam/hv3-comms/
- Add support for toast messages: https://github.com/adamstep/contact-app/commits/adam/hv3-messages/
- Add swipe gesture support: https://github.com/adamstep/contact-app/commits/adam/hv3-swipe/

**Comment by ramiboutas** [comment]
Many thanks for your response, @adamstep !

I cloned your basic contacts app repository and I still have the same issue. I checked the received xml with proxyman and it seems ok. See screenshots below.

![image](https://github.com/Instawork/hyperview/assets/33960822/a0934e03-fc01-4d98-8a08-aa4d7274cb4a)

![image](https://github.com/Instawork/hyperview/assets/33960822/4631484c-3a88-40b0-9bae-6ecb1e013222)

I tested this with 4 different systems: Ubuntu 22.04, Windows 11, Iphone and Android. And I get allways the same results.

I have never done anything with react native, but is there anything I can modify in the demo app in order to make it work?

Thanks a lot!
